source/agora/utils/Test.d

```diff
@@ -589,3 +591,174 @@ unittest
+        assert(0, "Invalid transaction passed to `split`");
+
+    auto remainder = amount.div(cast(uint) toward.length);
+    Transaction result = Transaction(type);
+    foreach (addr; toward)
+        result.outputs ~= Output(amount, addr);
+
+    // Add the remainder to the first output.
+    result.outputs[0].value.mustAdd(remainder);
+    const inputHash = input.hashFull();
+
+    // Add support for Transactions with multiple recipients
+    foreach (idx, const ref _; input.outputs)
+        result.inputs ~= Input(inputHash, cast(uint) idx);
+
+    auto sign = (KeyPair kp, Hash h) @trusted { return kp.secret.sign(h[]); };
+
+    const resultHash = result.hashFull();
+    foreach (idx, ref in_; result.inputs)
+    {
+        auto rng = from.find!(a => a.address == input.outputs[idx].address);
+        assert(rng.length);
+        const owner = rng[0];
+        in_.signature = sign(owner, resultHash);
+    }
+    return result;
+}
+
+/// Test for a split with the same amount of outputs as inputs
+/// Essentially doing an equality transformation
+unittest
+{
+    import std.range;
+
+    KeyPair[] keys = iota(8).map!(_ => KeyPair.random()).array;
+    KeyPair genesisKP = getGenesisKeyPair();
+    const first = GenesisBlock.txs[0];
+    const equalTx = first.split([genesisKP], keys.map!(k => k.address).array);
+    // This transaction has 8 txs, hence it's just equality
+    assert(equalTx.inputs.length == 8);
+    assert(equalTx.outputs.length == 8);
+    // Since the amount is evenly distributed in Genesis,
+    // they all have the same value
+    const ExpectedAmount = first.outputs[0].value;
+    assert(equalTx.outputs.all!(val => val.value == ExpectedAmount));
+}
+
+/// Test with twice as many outputs as inputs
+unittest
+{
+    import std.range;
+
+    KeyPair[] keys = iota(16).map!(_ => KeyPair.random()).array;
+    // Use Genesis
+    KeyPair genesisKP = getGenesisKeyPair();
+    const first = GenesisBlock.txs[0];
+    const equalTx = first.split([genesisKP], keys.map!(k => k.address).array);
+    // This transaction has 16 txs
+    assert(equalTx.inputs.length == 8);
+    assert(equalTx.outputs.length == 16);
+
+    Amount ExpectedAmount;
+    first.getSumOutput(ExpectedAmount);
+    auto remainder = ExpectedAmount.div(16);
+    assert(remainder == Amount(0));
+    assert(equalTx.outputs.all!(val => val.value == ExpectedAmount));
+}
+
+/// Test with remainder
+unittest
+{
+    import std.range;
+
+    KeyPair[] keys = iota(3).map!(_ => KeyPair.random()).array;
+    // Use Genesis
+    KeyPair genesisKP = getGenesisKeyPair();
+    const first = GenesisBlock.txs[0];
+    const equalTx = first.split([genesisKP], keys.map!(k => k.address).array);
+    // This transaction has 3 txs
+    assert(equalTx.inputs.length == 8);
+    assert(equalTx.outputs.length == 3);
+
+    Amount ExpectedAmount;
+    first.getSumOutput(ExpectedAmount);
+    auto remainder = ExpectedAmount.div(3);
+    assert(remainder != Amount(0));
```

Comment:
Using `getSumOutput` isn't great here. I'd recommend to hardcode the value (even if it means changing it later on).  One thing we don't want is having tests that replicate the code in the function. We should test the behavior instead, and having hard-coded value makes this simpler.

For example:
```suggestion
    // 500M / 3
    const Amount ExpectedAmount        = Amount(166_666_666_6666_666);
    const Amount ExpectedFirstAmount = Amount(166_666_666_6666_668);
```